compliance/collection/main.go

```diff
@@ -256,14 +262,25 @@ func main() {
 	// Set the random seed based on the current time.
 	rand.Seed(time.Now().UnixNano())
 
+	var niClient scannerV1.NodeInventoryServiceClient
+
 	if features.RHCOSNodeScanning.Enabled() {
 		// Start the prometheus metrics server
 		metrics.NewDefaultHTTPServer().RunForever()
 		metrics.GatherThrottleMetricsForever(metrics.ComplianceSubsystem.String())
+
+		// Set up Compliance <-> NodeInventory connection
+		niConn, err := clientconn.AuthenticatedGRPCConnection(nodeScannerEndpoint, mtls.Subject{}, clientconn.UseInsecureNoTLS(true))
+		if err != nil {
+			log.Fatal(err)
```

Comment:
This shouldn't happen. Compliance shouldn't go into crash loop if node inventory is not up yet or (more likely) misconfigured. If `ROX_NODE_NAME` is not set for node inventory, it will make compliance crash, which is the last thing we want. 

Even if you don't have the retry loop (yet), this should at least defer the start of the client until the next 4 hour timer kicks in (in manageNodeScanLoop).